app/emails/seller_edit.py

```diff
@@ -0,0 +1,102 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+import pendulum
+from flask import current_app
+from .util import render_email_template, send_or_handle_error, escape_markdown
+
+
+def send_notify_auth_rep_email(supplier_code):
+    from app.api.services import (
+        audit_service,
+        audit_types,
+        suppliers,
+        key_values_service,
+        agreement_service
+    )
+
+    supplier = suppliers.get_supplier_by_code(supplier_code)
+    to_address = supplier.data.get('email', '')
+
+    key_value = key_values_service.get_by_key('current_master_agreement')
+    start_date = pendulum.now('Australia/Canberra').date()
+    if key_value:
+        agreement = agreement_service.find(is_current=True).one_or_none()
+        agreement_sup = key_value.get('data').get('{}'.format(agreement.id))
+        start_date = (
+            pendulum.parse(
+                agreement_sup.get('startDate'),
+                tz='Australia/Canberra'
+            ).date()
+        )
+
+    # prepare copy
+    email_body = render_email_template(
+        'seller_edit_notify_auth_rep.md',
+        ma_start_date=start_date.strftime('%-d %B %Y'),
+        supplier_name=supplier.name,
+        auth_rep_name=escape_markdown(supplier.data.get('representative', '')),
+        frontend_url=current_app.config['FRONTEND_ADDRESS']
+    )
+
+    subject = "Accept the Master Agreement for the Digital Marketplace"
+
+    send_or_handle_error(
+        to_address,
+        email_body,
+        subject,
+        current_app.config['DM_GENERIC_NOREPLY_EMAIL'],
+        current_app.config['DM_GENERIC_SUPPORT_NAME'],
+        event_description_for_errors='notify auth rep email'
+    )
+
+    audit_service.log_audit_event(
+        audit_type=audit_types.notify_auth_rep_accept_master_agreement,
+        user='',
+        data={
+            "to_address": to_address,
+            "email_body": email_body,
+            "subject": subject
+        },
+        db_object=supplier)
+
+
+def send_decline_master_agreement_email(supplier_code):
+    from app.api.services import (
+        audit_service,
+        audit_types,
+        suppliers
+    )
+
+    supplier = suppliers.get_supplier_by_code(supplier_code)
+    to_addresses = [
+        e['email_address']
+        for e in suppliers.get_supplier_contacts(supplier_code)
+    ]
+
+    # prepare copy
+    email_body = render_email_template(
+        'seller_edit_decline.md',
+        frontend_url=current_app.config['FRONTEND_ADDRESS']
+    )
+
+    subject = "You declined the new Master Agreement"
+
+    send_or_handle_error(
+        to_addresses,
+        email_body,
+        subject,
+        current_app.config['DM_GENERIC_NOREPLY_EMAIL'],
+        current_app.config['DM_GENERIC_SUPPORT_NAME'],
+        event_description_for_errors='declined master agreement email'
+    )
+
+    audit_service.log_audit_event(
```

Comment:
Looks like some of these `audit_service` calls are wrapped in `try`.  Add or remove as you see fit!

Reply:
remove